### PR TITLE
{Core} Knack adoption

### PR DESCRIPTION
**Pull request by jiasli**
**History Notes:**  

[Core] PREVIEW: Allow disabling color by setting `AZURE_CORE_NO_COLOR` environment variable to `True` or `[core] no_color=True` config (#12601)
[Core] PREVIEW: Add `--only-show-errors` global argument to mute all warning, info and debug output. It can also be enabled by setting `AZURE_CORE_ONLY_SHOW_ERRORS` environment variable to `True` or `[core] only_show_errors=True` config (#12544)
[Core] PREVIEW: Add experimental tag to command groups, commands and arguments (#12543)
{Core} Move yaml output to Knack (#12603)

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S167

File: src/azure-cli-core/azure/cli/core/_help.py [hunk 1]
```diff
@@ -162,7 +166,8 @@ def show_help(self, cli_name, nouns, parser, is_group):
             AzCliHelp.update_examples(help_file)
         self._print_detailed_help(cli_name, help_file)
 
-        print(SURVEY_PROMPT)
+        from colorama import Fore, Style
```
Comment (qianwens) [inline src/azure-cli-core/azure/cli/core/_help.py]: >from colorama import Fore, Style [](start = 8, length = 32)

suggest to add some description in PR about colorama changes, I cannot understand the difference between colorama.init(autoreset=True) and use from colorama import Fore, Style

File: src/azure-cli-core/azure/cli/core/commands/constants.py [hunk 2]
```diff
@@ -31,5 +32,6 @@
 
 BLACKLISTED_MODS = ['context', 'shell', 'documentdb', 'component']
 
-SURVEY_PROMPT = Fore.YELLOW + Style.BRIGHT + 'Please let us know how we are doing: ' + Fore.BLUE \
-    + 'https://aka.ms/clihats' + Style.RESET_ALL
+SURVEY_PROMPT = 'Please let us know how we are doing: https://aka.ms/clihats'
+SURVEY_PROMPT_COLOR = Fore.YELLOW + Style.BRIGHT + 'Please let us know how we are doing: ' + Fore.BLUE \
```
Comment (qianwens) [inline src/azure-cli-core/azure/cli/core/commands/constants.py]: you can add some snapshot in PR description for this kind of ux change

**Comment by haroldrandom** [comment]
Could resolve conflict and errors first?

**Comment by jiasli** [comment]
> Could resolve conflict and errors first?

This is because Knack is not released and the dependency is missing. Could you please take a look at https://github.com/microsoft/knack/pull/181?

File: src/azure-cli-core/azure/cli/core/_help.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_help.py]: This is really a long store about terminal coloring. You may check the colorama doc https://pypi.org/project/colorama/ for its usage. `colorama.init(autoreset=True)` logic has been moved to Knack. 

This line is redundant. Let's remove it.

File: src/azure-cli-core/azure/cli/core/commands/constants.py (hunk 2 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/commands/constants.py]: Nice idea!

Originally with color:

image

After running `$Env:AZURE_CORE_NO_COLOR='true'`. It becomes 

image

**Pull request merged**